src/seedu/addressbook/TextUi.java

```diff
@@ -0,0 +1,327 @@
+
+    public static final String COMMAND_LIST_DESC = "Displays all persons as a list with index numbers.";
+    public static final String COMMAND_LIST_EXAMPLE = Command.LIST.commandWord;
+
+    public static final String COMMAND_VIEW_DESC = "Views the non-private details of the person identified by the "
+            + "index number shown in the last person listing.";
+    public static final String COMMAND_VIEW_PARAMETERS = "INDEX";
+    public static final String COMMAND_VIEW_EXAMPLE = Command.VIEW.commandWord + " 2";
+
+    public static final String COMMAND_VIEW_ALL_DESC = "Views all details of the person identified by the index number"
+            + " shown in the last person listing.";
+    public static final String COMMAND_VIEW_ALL_PARAMETERS = "INDEX";
+    public static final String COMMAND_VIEW_ALL_EXAMPLE = Command.VIEW_ALL.commandWord + " 3";
+
+    public static final String COMMAND_DELETE_DESC = "Deletes the person identified by the index number used in "
+            + "the last person listing.";
+    public static final String COMMAND_DELETE_PARAMETERS = "INDEX";
+    public static final String COMMAND_DELETE_EXAMPLE = Command.DELETE.commandWord + " 1";
+
+    public static final String COMMAND_CLEAR_DESC = "Clears address book permanently.";
+    public static final String COMMAND_CLEAR_EXAMPLE = Command.CLEAR.commandWord;
+
+    public static final String COMMAND_HELP_DESC = "Shows program usage instructions.";
+    public static final String COMMAND_HELP_EXAMPLE = Command.HELP.commandWord;
+
+    public static final String COMMAND_EXIT_DESC = "Exits the program.";
+    public static final String COMMAND_EXIT_EXAMPLE = Command.EXIT.commandWord;
+
+    /**
+     * Offset required to convert between 1-indexing and 0-indexing.COMMAND_
+     */
+    public static final int DISPLAYED_INDEX_OFFSET = 1;
+
+    private final Scanner in;
+    private final PrintStream out;
+    private final Parser parser;
+    
+    /**
+     * preserves the last showed listing of persons for understanding user person references
+     * based on the the last listing they saw
+     */
+    private List<? extends ReadOnlyPerson> lastShownPersonListing = new ArrayList<>();
+
+    /**
+     * latest input line retrieved from {@link #getUserCommand()}
+     */
+    private String lastEnteredCommand = "";
+    
+    public TextUi(InputStream in, PrintStream out, Parser parser) {
+        this.in = new Scanner(in);
+        this.out = out;
+        this.parser = parser;
+    }
+
+    /**
+     * Prompts for the command and reads the text entered by the user.
+     * Ignores empty, pure whitespace, and comment lines.
+     *
+     * @see Parser#shouldIgnore(String)
+     * @return full line entered by the user
+     */
+    public String getUserCommand() {
+        out.print(LINE_PREFIX + "Enter command: ");
+        String fullInputLine = in.nextLine();
+        // silently consume all ignored lines
+        while (parser.shouldIgnore(fullInputLine)) {
```

Comment:
This line seems to be the only reason for TextUi to know the Parser. See if you can get rid of it. You can move the `shouldIgnore()` method to the TextUi itself.